dissect/target/plugins/os/windows/_tasks_job.py

```diff
@@ -0,0 +1,508 @@
+import datetime
+import warnings
+from typing import Iterator, Optional
+
+from dissect import cstruct
+from flow.record import GroupedRecord
+
+from dissect.target.exceptions import InvalidTaskError
+from dissect.target.helpers.fsutil import TargetPath
+from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.target import Target
+
+warnings.simplefilter(action="ignore", category=FutureWarning)
+
+ExecRecord = TargetRecordDescriptor(
+    "filesystem/windows/task/action/exec",
+    [
+        ("string", "action_type"),
+        ("string", "command"),
+        ("string", "arguments"),
+        ("string", "working_directory"),
+    ],
+)
+
+TriggerRecord = TargetRecordDescriptor(
+    "filesystem/windows/task/trigger",
+    [
+        ("string", "enabled"),
+        ("string", "start_boundary"),
+        ("string", "end_boundary"),
+        ("string", "repetition_interval"),
+        ("string", "repetition_duration"),
+        ("string", "repetition_stop_duration_end"),
+        ("string", "execution_time_limit"),
+    ],
+)
+
+DailyTriggerRecord = TargetRecordDescriptor(
+    "filesystem/windows/task/trigger/daily",
+    [
+        ("uint16", "days_between_triggers"),
+        ("uint16[]", "unused"),
+    ],
+)
+
+WeeklyTriggerRecord = TargetRecordDescriptor(
+    "filesystem/windows/task/trigger/weekly",
+    [
+        ("uint16", "weeks_between_triggers"),
+        ("string[]", "days_of_week"),
+        ("uint16[]", "unused"),
+    ],
+)
+
+MonthlyDateTriggerRecord = TargetRecordDescriptor(
+    "filesystem/windows/task/trigger/monthly_date",
+    [
+        ("string", "day_of_month"),
+        ("string[]", "months_of_year"),
+    ],
+)
+
+MonthlyDowTriggerRecord = TargetRecordDescriptor(
+    "filesystem/windows/task/trigger/monthly_dow",
+    [
+        ("string", "which_week"),
+        ("string[]", "day_of_week"),
+        ("string", "months_of_year"),
+    ],
+)
+
+PaddingTriggerRecord = TargetRecordDescriptor(
+    "filesystem/windows/task/trigger/padding",
+    [
+        ("uint16", "padding"),
+        ("uint16", "reserved2"),
+        ("uint16", "reserved3"),
+    ],
+)
```

Comment:
All these records are shared between `_tasks_job.py` and `_tasks_xml.py`. I think it is better to split them out into something like `_tasks_records.py`.